src/renderer/services/ethereum/fees.ts

```diff
@@ -1,80 +1,82 @@
-  params: Array<any>
-): C.FeesLD =>
-  Rx.combineLatest([client$]).pipe(
-    RxOp.switchMap(([oClient]) =>
-      FP.pipe(
-        oClient,
-        O.fold(
-          () => Rx.EMPTY,
-          (client) =>
-            Rx.combineLatest([client.estimateCall(address, abi, func, params), client.estimateGasPrices()]).pipe(
-              RxOp.map(
-                ([gasLimit, gasPrices]) =>
-                  ({
-                    type: 'byte',
-                    average: getFee({ gasPrice: gasPrices.average, gasLimit }),
-                    fast: getFee({ gasPrice: gasPrices.fast, gasLimit }),
-                    fastest: getFee({ gasPrice: gasPrices.fastest, gasLimit })
-                  } as Fees)
+export const createFeesService = ({ client$, chain }: { client$: Client$; chain: Chain }): FeesService => {
+  const common = C.createFeesService<FeesParams>({ client$, chain })
+
+  const callFees$ = ({ address, abi, func, params }: CallFeeParams): C.FeesLD =>
+    client$.pipe(
+      RxOp.switchMap((oClient) =>
+        FP.pipe(
+          oClient,
+          O.fold(
+            () => Rx.EMPTY,
+            (client) =>
+              Rx.combineLatest([client.estimateCall(address, abi, func, params), client.estimateGasPrices()]).pipe(
+                RxOp.map(
+                  ([gasLimit, gasPrices]) =>
+                    ({
+                      type: 'byte',
+                      average: getFee({ gasPrice: gasPrices.average, gasLimit }),
+                      fast: getFee({ gasPrice: gasPrices.fast, gasLimit }),
+                      fastest: getFee({ gasPrice: gasPrices.fastest, gasLimit })
+                    } as Fees)
+                )
               )
-            )
+          )
         )
-      )
-    ),
-    RxOp.map(RD.success),
-    RxOp.catchError((_) => Rx.of(RD.success(getDefaultFees()))),
-    RxOp.startWith(RD.pending)
-  )
+      ),
+      RxOp.map(RD.success),
+      RxOp.catchError((_) => Rx.of(RD.success(getDefaultFees()))),
+      RxOp.startWith(RD.pending)
+    )
 
-export const outTxFee$ = (asset: Asset): C.FeesLD =>
-  Rx.combineLatest([client$]).pipe(
-    RxOp.switchMap(([oClient]) =>
-      FP.pipe(
-        oClient,
-        O.fold(
-          () => Rx.EMPTY,
-          (client) => {
-            const gasLimit = isEthAsset(asset) ? ETH_OUT_TX_GAS_LIMIT : ERC20_OUT_TX_GAS_LIMIT
-            return Rx.from(client.estimateGasPrices()).pipe(
-              RxOp.map(
-                (gasPrices) =>
-                  ({
-                    type: 'byte',
-                    average: getFee({ gasPrice: gasPrices.average, gasLimit }),
-                    fast: getFee({ gasPrice: gasPrices.fast, gasLimit }),
-                    fastest: getFee({ gasPrice: gasPrices.fastest, gasLimit })
-                  } as Fees)
+  const outTxFee$ = (asset: Asset): C.FeesLD =>
+    client$.pipe(
+      RxOp.switchMap((oClient) =>
+        FP.pipe(
+          oClient,
+          O.fold(
+            () => Rx.EMPTY,
```

Comment:
`Rx.EMPTY` -> `Rx.of(RD.initial)` - we will complete the stream in other case (see [empty](https://www.learnrxjs.io/learn-rxjs/operators/creation/empty#observable-that-immediately-completes) or [EMPTY](https://rxjs.dev/api/index/const/EMPTY))